src/devices/fakeBattery/fakeIMU.cpp

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include <string>
+#include <yarp/os/Thread.h>
+#include <yarp/os/Time.h>
+#include <yarp/os/Semaphore.h>
+#include <yarp/os/Stamp.h>
+#include <yarp/os/LogStream.h>
+
+#include <fakeIMU.h>
+
+using namespace yarp::os;
+using namespace yarp::dev;
+using namespace yarp::sig;
+using namespace yarp::math;
+
+/**
+ * This device implements a fake analog sensor
+ * emulating an IMU
+ * @author Alberto Cardellino
+ */
+fakeIMU::fakeIMU() : PeriodicThread(DEFAULT_PERIOD)
+{
+    nchannels = 12;
+    dummy_value = 0;
+    rpy.resize(3);
+    dcm.resize(4,4);
+    gravity.resize(4);
+    accels.resize(4);
+    rpy.zero();
+    dcm.zero();
+    accels.zero();
+
+    gravity[0] = 0.0;
+    gravity[1] = 0;
+    gravity[2] = -9.81;
+    gravity[3] = 0;
+
+    m_sensorName = "sensorName";
+    m_frameName  = "frameName";
+}
+
+fakeIMU::~fakeIMU()
+{
+    close();
+}
+
+bool fakeIMU::open(yarp::os::Searchable &config)
+{
+    double period;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'period' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)